keras_nlp/layers/multi_segment_packer.py

```diff
@@ -199,22 +205,20 @@ def _combine_inputs(self, segments):
     def call(self, inputs):
         inputs = self._sanitize_inputs(inputs)
 
-        # If rank 1, add a batch dim and convert to ragged.
+        # If rank 1, add a batch dim.
         rank_1 = inputs[0].shape.rank == 1
         if rank_1:
             inputs = [tf.expand_dims(x, 0) for x in inputs]
-            inputs = [tf.RaggedTensor.from_tensor(x) for x in inputs]
+        inputs = self._convert_inputs(inputs)
```

Comment:
Might be slight easier/readable keep the loop outside.

`inputs = [self._convert_dense(x) for x in input]`

```
def _convert_dense(input):
    if isinstance(input, tf.Tensor):
        return tf.RaggedTensor.from_tensor(input)
    return input
```